types/src/transaction/metadata.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
```

Comment:
Maybe add some module-level explanations here with `//! something ...` and use doc comments `///` below for each struct/enum.

You can check the result with `cargo doc -p libra-types --open`

Reply:
Btw. are these metadata only used by the Move script `PeerToPeerWithMetadata`? Maybe add cross references from the doc? https://github.com/libra/libra/blob/master/language/stdlib/transaction_scripts/peer_to_peer_with_metadata.move#L5